README.md

```diff
@@ -1,122 +1,170 @@
 
-Then we'll create two replacement files to reach into the true entry-points:
-
-```ts
-// index.ios.js
+React Native looks for the entry-point `index.js`. Move this file to `src/index.js`.
 
-import './src/index.ios';
+```
+mv index.js src
 ```
 
-```ts
-// index.android.js
+Create a replacement `index.js` where React Native is expecting the entry point, and have it `import` our entry point in `src`.
+
+```js
+// index.js
+
+import './src/index';
+```
+React Native initialization creates the top component App.js.
+Move this file into `src/App.js`.
 
-import './src/index.android';
+```sh
+mv App.js src
 ```
 
-We'll also move our `__tests__` directory into `src` as well.
+Move the `__tests__` directory into `src`.
 
 ```sh
 mv ./__tests__/ ./src/__tests__/
 ```
 
-Ensure that everything is working correctly.
-Try to deploy to a device with one of the two commands:
+#### updated directory structure
+
+```sh
+C:.
+│   index.js
+│   package.json
+│   yarn.lock
+|   ...
+|
+├───android
+├───ios
+├───src
+|   |   App.js
+|   |   index.js
+|   |
+|   └───__tests__
+|           App.js
+```
+
+To ensure that everything is working correctly, try to deploy to a device or emulator with one of the two commands:
 
 ```sh
 react-native run-android
 react-native run-ios
 ```
 
-And ensure your tests are still passing.
+Ensure your tests are still passing.
 
 ```sh
 npm test
 ```
 
-If all is still working, it'd be a good idea to commit our changes in some version control system like git before we introduce TypeScript.
-
-## Introducing TypeScript
-
-It's time to introduce TypeScript to our project.
-First, rewrite the root `index.ios.js` and `index.android.js` files to import from `lib` instead of `src`.
+If all is still working, it would be a good idea to commit the changes to a version control system before we introduce TypeScript.
 
-```ts
-// index.ios.js
+# Adding TypeScript
 
-import './lib/index.ios';
-```
+First, rewrite the root `index.js` files to import from `lib` instead of `src`.  `lib` will contin the compiled `tsc` output.
 
 ```ts
-// index.android.js
+// index.js
 
-import './lib/index.android';
+import './lib/index';
 ```
 
+### Adding the TypeScript configuration file
 
-### Adding a configuration file
-
-Let's create a `tsconfig.json`:
+Create a `tsconfig.json` that outputs to `lib` with the following command:
 
 ```ts
 tsc --init --pretty --sourceMap --target es2015 --outDir ./lib --module commonjs --jsx react
 ```
 
-We'll also need to add `./src/` to the `"include"` section of our `tsconfig.json`:
+Update the following in the`tsconfig.json:
 
 ```json
 {
     "compilerOptions": {
         // other options here
+         "types": ["react","react-native","jest"],
+         "allowSyntheticDefaultImports": true, 
+         "esModuleInterop": true

```

Comment:
I am not sure if any one else has ran into some issue when the Typescript compiler.

I believe  **"skipLibCheck": true,** should be added. to avoid this issue: **['types/react-native'] Property 'geolocation' must be of type 'Geolocation', but here has type 'GeolocationStatic'.**